protocol/writing-resource/containment.feature

```diff
@@ -52,3 +52,56 @@ Feature: Creating a resource using PUT and PATCH must create intermediate contai
     When method GET
     Then status 200
     And match testContainer.parseMembers(response) contains intermediateContainer.url
+
+  Scenario: PUT conflicts when creating resource turning resource into container
+    * def requestUri = rootTestContainer.url + 'dahut'
+    Given url requestUri
+    And configure headers = clients.alice.getAuthHeaders('PUT', requestUri)
+    And header Content-Type = 'text/turtle'
+    And request '<> a <http://example.org/Dahut> .'
+    When method PUT
+    Then assert responseStatus >= 200 && responseStatus < 300
+
+    * def childrenRequestUri = rootTestContainer.url + 'dahut/foo/bar.txt'
+    Given url childrenRequestUri
+    And configure headers = clients.alice.getAuthHeaders('PUT', childrenRequestUri)
+    And header Content-Type = 'text/plain'
+    And request 'Hello'
+    When method PUT
+    Then assert responseStatus >= 400 && responseStatus < 500
+
+  Scenario: PATCH conflicts when creating resource turning resource into container
+    * def requestUri = rootTestContainer.url + 'dahut2'
+    Given url requestUri
+    And configure headers = clients.alice.getAuthHeaders('PATCH', requestUri)
+    And header Content-Type = 'application/sparql-update'
+    And request 'INSERT DATA { <#hello> <#linked> <#world> . }'
+    When method PATCH
+    Then assert responseStatus >= 200 && responseStatus < 300
+
+    * def childrenRequestUri = rootTestContainer.url + 'dahut2/foo/bar.txt'
+    Given url childrenRequestUri
+    And configure headers = clients.alice.getAuthHeaders('PATCH', childrenRequestUri)
+    And header Content-Type = 'text/plain'
+    And request 'Hello'
+    When method PATCH
+    Then assert responseStatus >= 400 && responseStatus < 500
+
+  Scenario: POST should not create resource turning resource into container
+    * def requestUri = rootTestContainer.url + 'dahut3'
+    Given url requestUri
+    And configure headers = clients.alice.getAuthHeaders('PUT', requestUri)
+    And header Content-Type = 'text/turtle'
+    And request '<> a <http://example.org/Dahut> .'
+    When method PUT
+    Then assert responseStatus >= 200 && responseStatus < 300
+
+    * def childContainerRequestUri = rootTestContainer.url + 'dahut3/foo/'
+    Given url childContainerRequestUri
+    And configure headers = clients.alice.getAuthHeaders('POST', childContainerRequestUri)
+    And header Content-Type = 'text/turtle'
+    And request '<> a <http://example.org/Dahut-3> .'
+    When method POST
+    Then status 404
```

Comment:
I hadn't thought this variant before now. I understand why this test should respond 404, but can a POST ever create something in a non-existing container? What if it did POST with a slug of `foo/bar`?

What if we had a test that attempted to POST a new container (with a `ldp:BasicContainer` link) to `dahut3/` - that is a POST to an existing resource but would it fail with a resource type conflict and would that test belong here or elsewhere?

As it stands, should this last scenario be in a separate feature for the `sopr:server-post-target-not-found` requirement?